posydon/popsyn/synthetic_population.py

```diff
@@ -992,16 +992,38 @@ def __init__(
                 Pwarn(f"{filename} already contains a mass_per_metallicity "
                       "table. Overwriting the table!", "OverwriteWarning")
 
-            simulated_mass = np.sum(self.oneline[["S1_mass_i", "S2_mass_i"]].to_numpy())
+            try:
+        # Filter the data based on the condition
+                filtered_data = self.oneline[self.oneline["state_i"] == "initially_single_star"]
+                filtered_data1 = self.oneline[self.oneline["state_i"] != "initially_single_star"]
```

Comment:
Why don't you call them `filtered_data_binaries` and `filtered_data_single` as you do later for the masses?